lib/logstash/outputs/datadog_metrics.rb

```diff
@@ -116,8 +107,13 @@ def flush(events, final=false)
   end # def flush
 
   private
+
+  def series_to_json(series)
```

Comment:
The purpose is two-fold.  1) to isolate the json generation, makes it easier to change the implementation from h.to_json to Json.dump(h) or other mechanism. 2) Because the call is to a collaborator, it is easier to handle exceptions if they are expected.
Using a method in these scenarios allows us to be more precise about what is returned for the request body.
Yes, at runtime it will probably be JIT'ed inline so here its function is more about code layout.